--- a/setup.py	2018-05-06 23:25:08.757522119 +0200
+++ b/setup.py	2018-05-06 23:25:30.269797365 +0200
@@ -15,7 +15,6 @@
 
 if sys.platform.startswith('linux'):
     define_macros = [('HAVE_CLOCK_GETTIME', '1'),
-                     ('HAVE_LIBRT', '1'),
                      ('HAVE_POSIX_MEMALIGN', '1'),
                      ('HAVE_STRUCT_SYSINFO', '1'),
                      ('HAVE_STRUCT_SYSINFO_MEM_UNIT', '1'),
@@ -23,8 +22,7 @@
                      ('HAVE_SYSINFO', '1'),
                      ('HAVE_SYS_SYSINFO_H', '1'),
                      ('_FILE_OFFSET_BITS', '64')]
-    libraries = ['crypto', 'rt']
-    includes = ['/usr/local/include', '/usr/include']
+    libraries = ['crypto']
     CFLAGS.append('-O2')
 elif sys.platform.startswith('win32'):
     define_macros = [('inline', '__inline')]
