--- netifaces/setup.py.orig	2018-05-02 09:45:09.000000000 +0200
+++ netifaces/setup.py	2018-12-11 14:12:02.785808692 +0100
@@ -55,7 +55,7 @@
         self.check_requirements()
         build_ext.build_extensions(self)
 
-    def test_build(self, contents, link=True, execute=False, libraries=None,
+    def test_build(self, contents, link=False, execute=False, libraries=None,
                    include_dirs=None, library_dirs=None):
         name = os.path.join(self.build_temp, 'conftest-%s.c' % self.conftestidx)
         self.conftestidx += 1
