--- a/Android.mk
+++ b/Android.mk
@@ -33,8 +33,8 @@
 
 # Enable this if you want to support loading WebP images
 # The library path should be a relative path to this directory.
-SUPPORT_WEBP ?= false
-SUPPORT_SAVE_WEBP ?= true
+SUPPORT_WEBP := true
+SUPPORT_SAVE_WEBP := true
 WEBP_LIBRARY_PATH := external/libwebp
 
 
@@ -160,7 +160,7 @@
 ifeq ($(SUPPORT_WEBP),true)
     LOCAL_C_INCLUDES += $(LOCAL_PATH)/$(WEBP_LIBRARY_PATH)/src
     LOCAL_CFLAGS += -DLOAD_WEBP
-    LOCAL_STATIC_LIBRARIES += webpdemux
+    LOCAL_STATIC_LIBRARIES += webpmux webpdemux
     LOCAL_STATIC_LIBRARIES += webp
 ifeq ($(SUPPORT_SAVE_WEBP),true)
     LOCAL_CFLAGS += -DSAVE_WEBP=1