diff --git a/vispy/util/fonts/_freetype.py b/vispy/util/fonts/_freetype.py
index 3b33d0b..229d559 100644
--- a/vispy/util/fonts/_freetype.py
+++ b/vispy/util/fonts/_freetype.py
@@ -12,12 +12,12 @@ import numpy as np
 
 # Convert face to filename
 from ._vispy_fonts import _vispy_fonts, _get_vispy_font_filename
-if sys.platform.startswith('linux'):
-    from ...ext.fontconfig import find_font
-elif sys.platform.startswith('win'):
-    from ._win32 import find_font  # noqa, analysis:ignore
-else:
-    raise NotImplementedError
+# if sys.platform.startswith('linux'):
+#     from ...ext.fontconfig import find_font
+# elif sys.platform.startswith('win'):
+#     from ._win32 import find_font  # noqa, analysis:ignore
+# else:
+#     raise NotImplementedError
 
 _font_dict = {}
 
@@ -41,6 +41,7 @@ def _load_font(face, bold, italic):
 
 def _load_glyph(f, char, glyphs_dict):
     """Load glyph from font into dict"""
+    return
     from ...ext.freetype import (FT_LOAD_RENDER, FT_LOAD_NO_HINTING,
                                  FT_LOAD_NO_AUTOHINT)
     flags = FT_LOAD_RENDER | FT_LOAD_NO_HINTING | FT_LOAD_NO_AUTOHINT
