--- Python-2.7.15/configure.orig	2018-04-30 00:47:33.000000000 +0200
+++ Python-2.7.15/configure	2019-08-16 14:09:48.696030207 +0200
@@ -2903,7 +2903,7 @@ case $host_os in *\ *) host_os=`echo "$h
 # pybuilddir.txt will be created by --generate-posix-vars in the Makefile
 rm -f pybuilddir.txt
 
-for ac_prog in python$PACKAGE_VERSION python3 python
+for ac_prog in python$PACKAGE_VERSION python2 python
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
