--- Python-2.7.9/Lib/distutils/command/build_ext.py.orig	2014-12-10 16:59:34.000000000 +0100
+++ Python-2.7.9/Lib/distutils/command/build_ext.py	2016-03-17 11:45:06.953243146 +0100
@@ -747,9 +747,9 @@
             # extensions, it is a reference to the original list
             return ext.libraries + [pythonlib, "m"] + extra
 
-        elif sys.platform == 'darwin':
-            # Don't use the default code below
-            return ext.libraries
+        # elif sys.platform == 'darwin':
+        #     # Don't use the default code below
+        #     return ext.libraries
         elif sys.platform[:3] == 'aix':
             # Don't use the default code below
             return ext.libraries
