--- Python-2.7.2/configure.orig	2015-06-24 17:47:39.181473779 +0200
+++ Python-2.7.2/configure	2015-06-24 17:48:31.646173137 +0200
@@ -5731,7 +5731,7 @@
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether gcc supports ParseTuple __format__" >&5
 $as_echo_n "checking whether gcc supports ParseTuple __format__... " >&6; }
   save_CFLAGS=$CFLAGS
-  CFLAGS="$CFLAGS -Werror"
+  CFLAGS="$CFLAGS -Werror -Wformat"
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
