--- Python-2.7.9/Makefile.pre.in.orig	2016-03-17 11:14:52.005098000 +0100
+++ Python-2.7.9/Makefile.pre.in	2016-03-17 12:59:54.490618508 +0100
@@ -470,12 +470,12 @@
 # Under BSD make, MAKEFLAGS might be " -s -v x=y".
 sharedmods: $(BUILDPYTHON) pybuilddir.txt
 	@case "$$MAKEFLAGS" in \
-	    *\ -s*|s*) quiet="-q";; \
+	    *\ -s*|s*) quiet="";; \
 	    *) quiet="";; \
 	esac; \
 	$(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' \
 		_TCLTK_INCLUDES='$(TCLTK_INCLUDES)' _TCLTK_LIBS='$(TCLTK_LIBS)' \
-		$(PYTHON_FOR_BUILD) $(srcdir)/setup.py $$quiet build
+		$(PYTHON_FOR_BUILD) $(srcdir)/setup.py $$quiet build -v
 
 # Build static library
 # avoid long command lines, same as LIBRARY_OBJS
