diff --git a/Android.mk b/Android.mk
index 97a96c7..2e724c0 100644
--- a/Android.mk
+++ b/Android.mk
@@ -79,6 +79,7 @@ ifeq ($(SUPPORT_JPG),true)
         $(JPG_LIBRARY_PATH)/jfdctfst.c \
         $(JPG_LIBRARY_PATH)/jfdctint.c \
         $(JPG_LIBRARY_PATH)/jidctflt.c \
+        $(JPG_LIBRARY_PATH)/jidctfst.c \
         $(JPG_LIBRARY_PATH)/jidctint.c \
         $(JPG_LIBRARY_PATH)/jquant1.c \
         $(JPG_LIBRARY_PATH)/jquant2.c \
@@ -86,12 +87,6 @@ ifeq ($(SUPPORT_JPG),true)
         $(JPG_LIBRARY_PATH)/jmemmgr.c \
         $(JPG_LIBRARY_PATH)/jmem-android.c
 
-    # assembler support is available for arm
-    ifeq ($(TARGET_ARCH),arm)
-        LOCAL_SRC_FILES += $(JPG_LIBRARY_PATH)/jidctfst.S
-    else
-        LOCAL_SRC_FILES += $(JPG_LIBRARY_PATH)/jidctfst.c
-    endif
 endif
 
 ifeq ($(SUPPORT_PNG),true)
diff --git a/external/jpeg-9/Android.mk b/external/jpeg-9/Android.mk
index a5edbde..77f139c 100644
--- a/external/jpeg-9/Android.mk
+++ b/external/jpeg-9/Android.mk
@@ -14,20 +14,6 @@ LOCAL_SRC_FILES := \
     jquant2.c jutils.c jmemmgr.c \
 	jmem-android.c
 
-# the assembler is only for the ARM version, don't break the Linux sim
-ifneq ($(TARGET_ARCH),arm)
-ANDROID_JPEG_NO_ASSEMBLER := true
-endif
-
-# temp fix until we understand why this broke cnn.com
-#ANDROID_JPEG_NO_ASSEMBLER := true
-
-ifeq ($(strip $(ANDROID_JPEG_NO_ASSEMBLER)),true)
-LOCAL_SRC_FILES += jidctint.c jidctfst.c
-else
-LOCAL_SRC_FILES += jidctint.c jidctfst.S
-endif
-
 LOCAL_CFLAGS += -DAVOID_TABLES 
 LOCAL_CFLAGS += -O3 -fstrict-aliasing -fprefetch-loop-arrays
 #LOCAL_CFLAGS += -march=armv6j
