diff -urN X3DH-0.5.3.ori/setup.py X3DH-0.5.3/setup.py
--- X3DH-0.5.3.ori/setup.py	2018-10-28 19:15:16.444766623 +0100
+++ X3DH-0.5.3/setup.py	2018-10-28 19:15:38.028060948 +0100
@@ -24,7 +24,7 @@
     author_email = "tim@cifg.io",
     license = "MIT",
     packages = find_packages(),
-    install_requires = [ "cryptography>=1.7.1", "XEdDSA>=0.4.2" ],
+    install_requires = [],
     python_requires  = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, <4",
     zip_safe = True,
     classifiers = [
