--- jpeg/CMakeLists.txt.orig	2018-11-12 20:20:28.000000000 +0100
+++ jpeg/CMakeLists.txt	2018-12-14 12:43:45.338704504 +0100
@@ -573,6 +573,9 @@
     add_library(turbojpeg SHARED ${TURBOJPEG_SOURCES})
     set_property(TARGET turbojpeg PROPERTY COMPILE_FLAGS
       "-DBMP_SUPPORTED -DPPM_SUPPORTED")
+    set_property(TARGET jpeg PROPERTY NO_SONAME 1)
+    set_property(TARGET turbojpeg PROPERTY NO_SONAME 1)
+    set(CMAKE_SHARED_LIBRARY_SONAME_C_FLAG "")
     if(WIN32)
       set_target_properties(turbojpeg PROPERTIES DEFINE_SYMBOL DLLDEFINE)
     endif()
