--- ../../../build/sqlalchemy/sqlalchemy-0.7/setup.py	2013-07-31 22:07:19.000000000 -0300
+++ setup.py	2013-08-10 02:35:53.530761349 -0300
@@ -10,17 +10,16 @@
 from distutils.command.build_ext import build_ext
 from distutils.errors import (CCompilerError, DistutilsExecError,
                               DistutilsPlatformError)
-try:
-    from setuptools import setup, Extension, Feature
-    has_setuptools = True
-except ImportError:
-    has_setuptools = False
-    from distutils.core import setup, Extension
-    Feature = None
-    try:  # Python 3
-        from distutils.command.build_py import build_py_2to3 as build_py
-    except ImportError:  # Python 2
-        from distutils.command.build_py import build_py
+
+has_setuptools = False
+from distutils.core import setup
+from distutils.extension import Extension
+Feature = None
+try:  # Python 3
+    from distutils.command.build_py import build_py_2to3 as build_py
+except ImportError:  # Python 2
+    from distutils.command.build_py import build_py
+
 
 cmdclass = {}
 pypy = hasattr(sys, 'pypy_version_info')
