diff --git a/third_party/cares/config_darwin/ares_config.h b/third_party/cares/config_darwin/ares_config.h
--- a/third_party/cares/config_darwin/ares_config.h	2024-07-16 20:46:22.000000000 +0100
+++ b/third_party/cares/config_darwin/ares_config.h	2024-07-29 00:18:30.096755745 +0100
@@ -43,7 +43,7 @@
 #define GETNAMEINFO_TYPE_ARG7 int
 
 /* Specifies the number of arguments to getservbyport_r */
-#define GETSERVBYPORT_R_ARGS 
+/* #define GETSERVBYPORT_R_ARGS */
 
 /* Define to 1 if you have AF_INET6. */
 #define HAVE_AF_INET6
diff --git a/third_party/cares/config_linux/ares_config.h b/third_party/cares/config_linux/ares_config.h
--- a/third_party/cares/config_linux/ares_config.h	2024-07-16 20:46:22.000000000 +0100
+++ b/third_party/cares/config_linux/ares_config.h	2024-07-29 00:19:39.479166654 +0100
@@ -43,7 +43,7 @@
 #define GETNAMEINFO_TYPE_ARG7 int
 
 /* Specifies the number of arguments to getservbyport_r */
-#define GETSERVBYPORT_R_ARGS 6
+/* #define GETSERVBYPORT_R_ARGS 6 */
 
 /* Define to 1 if you have AF_INET6. */
 #define HAVE_AF_INET6
@@ -121,7 +121,7 @@
 #define HAVE_GETNAMEINFO
 
 /* Define to 1 if you have the getservbyport_r function. */
-#define HAVE_GETSERVBYPORT_R
+/* #define HAVE_GETSERVBYPORT_R */
 
 /* Define to 1 if you have the `gettimeofday' function. */
 #define HAVE_GETTIMEOFDAY