--- a/src/CMakeLists.txt	2015-05-10 22:57:48.000000000 +0200
+++ b/src/CMakeLists.txt	2015-08-14 02:22:27.020473858 +0200
@@ -43,8 +43,10 @@
                text.cpp text.h
                utf8.cpp utf8.h
                util.cpp util.h
+               glob.c glob.h
                wcwidth6.cpp)
 
+add_library (glob glob.c)
 add_library (task STATIC ${task_SRCS})
 add_executable (task_executable main.cpp)
 add_executable (calc_executable calc.cpp)
@@ -64,6 +64,9 @@
 set_property (TARGET calc_executable PROPERTY OUTPUT_NAME "calc")
 set_property (TARGET lex_executable PROPERTY OUTPUT_NAME "lex")
 
+set (CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} "-fPIE")
+set (CMAKE_EXE_LINKER_FLAGS ${CMAKE_EXE_LINKER_FLAGS} "-pie")
+
 #SET(CMAKE_BUILD_TYPE gcov)
 #SET(CMAKE_CXX_FLAGS_GCOV "--coverage")
 #SET(CMAKE_C_FLAGS_GCOV "--coverage")
