--- leveldb/build_detect_platform	2014-09-16 23:19:52.000000000 +0200
+++ leveldb-patch/build_detect_platform	2016-03-01 21:56:04.926650079 +0100
@@ -156,7 +157,7 @@
 # except for the test and benchmark files. By default, find will output a list
 # of all files matching either rule, so we need to append -print to make the
 # prune take effect.
-DIRS="$PREFIX/db $PREFIX/util $PREFIX/table"
+DIRS="$PREFIX/snappy $PREFIX/db $PREFIX/util $PREFIX/table"
 
 set -f # temporarily disable globbing so that our patterns aren't expanded
 PRUNE_TEST="-name *test*.cc -prune"
