--- zope_interface/setup.py	2015-10-05 09:35:14.000000000 +0200
+++ b/setup.py	2016-06-15 17:44:35.108263993 +0200
@@ -139,9 +139,8 @@
         "Topic :: Software Development :: Libraries :: Python Modules",
       ],
 
-      packages = ['zope', 'zope.interface', 'zope.interface.tests'],
+      packages = ['zope', 'zope.interface'],
       package_dir = {'': 'src'},
       cmdclass = {'build_ext': optional_build_ext,
                   },
-      test_suite = 'zope.interface.tests',
       **extra)
