diff -Naur Python-2.7.9-orig/configure.ac Python-2.7.9/configure.ac
--- Python-2.7.9-orig/configure.ac	2014-12-11 13:49:36.865400000 +0300
+++ Python-2.7.9/configure.ac	2014-12-11 13:49:36.990200000 +0300
@@ -751,6 +751,27 @@
     ;;
 esac
 
+# initialize default configuration
+py_config=
+case $host in
+  *-*-mingw*) py_config=mingw ;;
+esac
+if test -n "$py_config" ; then
+  AC_MSG_NOTICE([loading configure defaults from .../Misc/config_$py_config"])
+  . "$srcdir/Misc/config_$py_config"
+fi
+
+# initialize defaults for cross-builds
+if test "$cross_compiling" = yes; then
+  py_config=$host_os
+  case $py_config in
+    mingw32*) py_config=mingw32 ;;
+  esac
+  if test -f "$srcdir/Misc/cross_$py_config" ; then
+    AC_MSG_NOTICE([loading cross defaults from .../Misc/cross_$py_config"])
+    . "$srcdir/Misc/cross_$py_config"
+  fi
+fi
 
 AC_SUBST(LIBRARY)
 AC_MSG_CHECKING(LIBRARY)
diff -Naur Python-2.7.9-orig/Misc/config_mingw Python-2.7.9/Misc/config_mingw
--- Python-2.7.9-orig/Misc/config_mingw	1970-01-01 03:00:00.000000000 +0300
+++ Python-2.7.9/Misc/config_mingw	2014-12-11 13:49:36.990200000 +0300
@@ -0,0 +1,12 @@
+# configure defaults for mingw* hosts
+
+# mingw functions to ignore
+ac_cv_func_ftruncate=ignore	# implement it as _chsize
+
+# mingw-w64 functions to ignore
+ac_cv_func_truncate=ignore
+ac_cv_func_alarm=ignore
+
+# files to ignore
+ac_cv_file__dev_ptmx=ignore #NOTE: under MSYS environment device exist
+ac_cv_file__dev_ptc=no
diff -Naur Python-2.7.9-orig/Misc/cross_mingw32 Python-2.7.9/Misc/cross_mingw32
--- Python-2.7.9-orig/Misc/cross_mingw32	1970-01-01 03:00:00.000000000 +0300
+++ Python-2.7.9/Misc/cross_mingw32	2014-12-11 13:49:37.037000000 +0300
@@ -0,0 +1,11 @@
+# configure defaults for mingw32 host if cross-build
+
+ac_cv_little_endian_double=yes
+ac_cv_big_endian_double=no
+ac_cv_mixed_endian_double=no
+
+ac_cv_tanh_preserves_zero_sign=yes
+
+ac_cv_wchar_t_signed=no
+
+ac_cv_have_size_t_format=no
