diff -urN a/Lib/ssl.py b/Lib/ssl.py
--- a/Lib/ssl.py	2014-10-11 14:23:27.721614900 +0100
+++ b/Lib/ssl.py	2014-10-11 14:23:30.169754900 +0100
@@ -144,7 +144,7 @@
     _SSLv2_IF_EXISTS = None
 
 from socket import socket, _fileobject, _delegate_methods, error as socket_error
-if sys.platform == "win32":
+if sys.platform == "win32" and sys.version.find("GCC") == -1:
     from _ssl import enum_certificates, enum_crls
 
 from socket import socket, AF_INET, SOCK_STREAM, create_connection
@@ -387,7 +387,7 @@
     def load_default_certs(self, purpose=Purpose.SERVER_AUTH):
         if not isinstance(purpose, _ASN1Object):
             raise TypeError(purpose)
-        if sys.platform == "win32":
+        if sys.platform == "win32" and sys.version.find("GCC") == -1:
             for storename in self._windows_cert_stores:
                 self._load_windows_store_certs(storename, purpose)
         else:
