diff -Naur Python-2.7.9-orig/Lib/distutils/cygwinccompiler.py Python-2.7.9/Lib/distutils/cygwinccompiler.py
--- Python-2.7.9-orig/Lib/distutils/cygwinccompiler.py	2014-12-11 13:50:54.147800000 +0300
+++ Python-2.7.9/Lib/distutils/cygwinccompiler.py	2014-12-11 13:51:21.245000000 +0300
@@ -330,9 +330,9 @@
         else:
             no_cygwin = ''
 
-        self.set_executables(compiler='gcc%s -O -Wall' % no_cygwin,
-                             compiler_so='gcc%s -mdll -O -Wall' % no_cygwin,
-                             compiler_cxx='g++%s -O -Wall' % no_cygwin,
+        self.set_executables(compiler='gcc%s -O2 -Wall' % no_cygwin,
+                             compiler_so='gcc%s -mdll -O2 -Wall' % no_cygwin,
+                             compiler_cxx='g++%s -O2 -Wall' % no_cygwin,
                              linker_exe='gcc%s' % no_cygwin,
                              linker_so='%s%s %s %s'
                                     % (self.linker_dll, no_cygwin,
