diff -Naur Python-2.7.9-orig/Modules/gdbmmodule.c Python-2.7.9/Modules/gdbmmodule.c
--- Python-2.7.9-orig/Modules/gdbmmodule.c	2014-12-10 18:59:55.000000000 +0300
+++ Python-2.7.9/Modules/gdbmmodule.c	2014-12-11 13:51:28.155800000 +0300
@@ -11,7 +11,7 @@
 #include <fcntl.h>
 #include "gdbm.h"
 
-#if defined(WIN32) && !defined(__CYGWIN__)
+#if defined(WIN32) && !defined(__CYGWIN__) && !defined(__MINGW32__)
 #include "gdbmerrno.h"
 extern const char * gdbm_strerror(gdbm_error);
 #endif
