diff -Nru Python-3.8.2/Lib/site.py Python-3.8.2-new/Lib/site.py
--- Python-3.8.2/Lib/site.py    2020-04-28 12:48:38.000000000 -0700
+++ Python-3.8.2-new/Lib/site.py        2020-04-28 12:52:46.000000000 -0700
@@ -487,7 +487,8 @@
                     if key == 'include-system-site-packages':
                         system_site = value.lower()
                     elif key == 'home':
-                        sys._home = value
+                        # this is breaking pyconfig.h path detection with venv
+                        print('Ignoring "sys._home = value" override', file=sys.stderr)
 
         sys.prefix = sys.exec_prefix = site_prefix
 
