diff -r 9d2a211310b3 -r 3ff3e1e49525 Python/dynload_shlib.c
--- a/Python/dynload_shlib.c	Mon May 07 00:04:24 2012 -0300
+++ b/Python/dynload_shlib.c	Mon May 07 00:04:47 2012 -0300
@@ -6,6 +6,7 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <stdlib.h>
 
 #if defined(__NetBSD__)
 #include <sys/param.h>
@@ -75,6 +76,21 @@
     char pathbuf[260];
     int dlopenflags=0;
 
+    static void *libpymodules = NULL;
+    void *rv = NULL;
+
+    /* Ensure we have access to libpymodules. */
+    if (libpymodules == NULL) {
+        printf("ANDROID_PRIVATE = %s\n", getenv("ANDROID_PRIVATE"));
+        PyOS_snprintf(pathbuf, sizeof(pathbuf), "%s/libpymodules.so", getenv("ANDROID_PRIVATE"));
+        libpymodules = dlopen(pathbuf, RTLD_NOW);
+
+        if (libpymodules == NULL) {
+            //abort();
+        }
+    }
+        
+
     if (strchr(pathname, '/') == NULL) {
         /* Prefix bare filename with "./" */
         PyOS_snprintf(pathbuf, sizeof(pathbuf), "./%-.255s", pathname);
@@ -84,6 +100,17 @@
     PyOS_snprintf(funcname, sizeof(funcname),
                   LEAD_UNDERSCORE "init%.200s", shortname);
 
+
+    /* Read symbols that have been linked into the main binary. */
+
+    if (libpymodules) {
+        rv = dlsym(libpymodules, funcname);
+        if (rv != NULL) {
+            return rv;
+        }
+    }
+
+
     if (fp != NULL) {
         int i;
         struct stat statb;
diff -r 9d2a211310b3 -r 3ff3e1e49525 Python/pythonrun.c
--- a/Python/pythonrun.c	Mon May 07 00:04:24 2012 -0300
+++ b/Python/pythonrun.c	Mon May 07 00:04:47 2012 -0300
@@ -261,9 +261,13 @@
     _PyGILState_Init(interp, tstate);
 #endif /* WITH_THREAD */
 
+    /* For PGS4A, we don't want to call initsite, as we won't have the
+       library path set up until start.pyx finishes running. */
+#if 0
     if (!Py_NoSiteFlag)
         initsite(); /* Module site */
-
+#endif 
+    
     if ((p = Py_GETENV("PYTHONIOENCODING")) && *p != '\0') {
         p = icodeset = codeset = strdup(p);
         free_codeset = 1;
