diff -r 2e01483f343a -r 0b997d01cabc setup.py
--- a/setup.py	Sun May 06 23:52:25 2012 -0300
+++ b/setup.py	Mon May 07 00:03:53 2012 -0300
@@ -445,6 +445,13 @@
                 '/lib', '/usr/lib',
                 ]
             inc_dirs += ['/usr/include']
+        else:
+            cflags = os.environ.get('CFLAGS')
+            if cflags:
+                inc_dirs += [x[2:] for x in cflags.split() if x.startswith('-I')]
+            ldflags = os.environ.get('LDFLAGS')
+            if ldflags:
+                lib_dirs += [x[2:] for x in ldflags.split() if x.startswith('-L')]
         exts = []
         missing = []
 
