diff --git a/jni/Android.mk b/jni/Android.mk
index df2ff1a..2f70985 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -26,8 +26,9 @@ LOCAL_SRC_FILES :=\
 	arm/filter_neon_intrinsics.c
 
 #LOCAL_SHARED_LIBRARIES := -lz
-LOCAL_EXPORT_LDLIBS := -lz
+# LOCAL_EXPORT_LDLIBS := -lz
+LOCAL_LDLIBS := -lz
 LOCAL_EXPORT_C_INCLUDES := $(LOCAL_PATH)/.
 
-#include $(BUILD_SHARED_LIBRARY)
-include $(BUILD_STATIC_LIBRARY)
+include $(BUILD_SHARED_LIBRARY)
+# include $(BUILD_STATIC_LIBRARY)
