diff -Naur a/python/lib/moresetuptools.py  b/python/lib/moresetuptools.py 

--- a/python/lib/moresetuptools.py   2024-02-01 22:35:52.713443658 -0500
+++ b/python/lib/moresetuptools.py.bkup      2024-02-02 00:54:22.274822119 -0500
@@ -179,7 +179,7 @@
                 add_local_aubio_lib(extension)
             else:
                 # check for external dependencies
-                add_external_deps(extension, usedouble=enable_double)
+                #add_external_deps(extension, usedouble=enable_double)
                 # force adding libav on windows
                 if os.name == 'nt' and ('WITH_LIBAV' in os.environ \
                         or 'CONDA_PREFIX' in os.environ):
