diff --git a/src/_cffi_src/build_openssl.py b/src/_cffi_src/build_openssl.py
index 7ec235ff..ba4f5c9d 100644
--- a/src/_cffi_src/build_openssl.py
+++ b/src/_cffi_src/build_openssl.py
@@ -33,7 +33,8 @@ def _get_openssl_libraries(platform):
         # specified on the linker command-line is significant;
         # libssl must come before libcrypto
         # (http://marc.info/?l=openssl-users&m=135361825921871)
-        return ["ssl", "crypto"]
+        version = os.environ.get("OPENSSL_VERSION", "")
+        return ["ssl" + version, "crypto" + version]
 
 
 def _extra_compile_args(platform):
