diff -Naur libffi/Android.mk b/Android.mk
--- libffi/Android.mk	2015-12-22 17:00:48.025478556 -0600
+++ b/Android.mk	2015-12-22 17:02:23.999249390 -0600
@@ -23,23 +23,20 @@
 # Build rules for the target.
 #
 
-# We only build ffi for mips.
-ifeq ($(TARGET_ARCH),mips)
 
-   include $(CLEAR_VARS)
+include $(CLEAR_VARS)
 
-   ffi_arch := $(TARGET_ARCH)
-   ffi_os := $(TARGET_OS)
+ffi_arch := $(TARGET_ARCH)
+ffi_os := $(TARGET_OS)
 
-   # This include just keeps the nesting a bit saner.
-   include $(LOCAL_PATH)/Libffi.mk
+# This include just keeps the nesting a bit saner.
+include $(LOCAL_PATH)/Libffi.mk
 
-   LOCAL_MODULE_TAGS := optional
-   LOCAL_MODULE := libffi
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE := libffi
 
-   include $(BUILD_SHARED_LIBRARY)
+include $(BUILD_SHARED_LIBRARY)
 
-endif
 
 # Also include the rules for the test suite.
 include external/libffi/testsuite/Android.mk
