diff -Naur orig/setup.py v0.4.7/setup.py
--- orig/setup.py	2015-06-11 14:16:31.315765908 -0500
+++ v0.4.7/setup.py	2015-06-11 14:17:05.800263536 -0500
@@ -64,7 +64,7 @@
 
 #-----------------------------------------------------------------------------
 def create_ecodes():
-    header = '/usr/include/linux/input.h'
+    header = os.environ['SYSROOT'] + '/usr/include/linux/input.h'
 
     if not os.path.isfile(header):
         msg = '''\
