--- a/websocket/_logging.py	2016-10-02 21:35:05.000000000 -0400
+++ b/websocket/_logging.py	2017-04-13 21:17:34.539598154 -0400
@@ -19,9 +19,10 @@
     Boston, MA  02110-1335  USA
 
 """
-import logging
-
-_logger = logging.getLogger('websocket')
+#import logging
+from kivy.logger import Logger
+#_logger = logging.getLogger('websocket')
+_logger = Logger
 _traceEnabled = False
 
 __all__ = ["enableTrace", "dump", "error", "debug", "trace",
@@ -67,8 +68,9 @@
 
 
 def isEnabledForError():
-    return _logger.isEnabledFor(logging.ERROR)
-
+#    return _logger.isEnabledFor(logging.ERROR)
+    return True
 
 def isEnabledForDebug():
-    return _logger.isEnabledFor(logging.DEBUG)
+#    return _logger.isEnabledFor(logging.DEBUG)
+    return True
