--- boost/boostcpp.jam	2015-12-14 03:30:09.000000000 +0100
+++ boost-patch/boostcpp.jam	2016-02-08 16:38:40.510859612 +0100
@@ -155,8 +155,9 @@
         if $(type) = SHARED_LIB &&
           ! [ $(property-set).get <target-os> ] in windows cygwin darwin aix &&
           ! [ $(property-set).get <toolset> ] in pgi
         {
+            return $(result) ; # disable version suffix for android
             result = $(result).$(BOOST_VERSION)  ;
         }
 
         return $(result) ;
