--- orig/Android.mk	2016-01-03 07:15:57.000000000 +0100
+++ patch/Android.mk	2016-04-15 21:28:55.169697882 +0200
@@ -6,22 +6,22 @@
 
 # Enable this if you want to support loading MOD music via modplug
 # The library path should be a relative path to this directory.
-SUPPORT_MOD_MODPLUG ?= true
+SUPPORT_MOD_MODPLUG := false
 MODPLUG_LIBRARY_PATH := external/libmodplug-0.8.8.4
 
 # Enable this if you want to support loading MOD music via mikmod
 # The library path should be a relative path to this directory.
-SUPPORT_MOD_MIKMOD ?= true
+SUPPORT_MOD_MIKMOD := false
 MIKMOD_LIBRARY_PATH := external/libmikmod-3.1.12
 
 # Enable this if you want to support loading MP3 music via SMPEG
 # The library path should be a relative path to this directory.
-SUPPORT_MP3_SMPEG ?= true
+SUPPORT_MP3_SMPEG := false
 SMPEG_LIBRARY_PATH := external/smpeg2-2.0.0
 
 # Enable this if you want to support loading OGG Vorbis music via Tremor
 # The library path should be a relative path to this directory.
-SUPPORT_OGG ?= true
+SUPPORT_OGG := true
 OGG_LIBRARY_PATH := external/libogg-1.3.1
 VORBIS_LIBRARY_PATH := external/libvorbisidec-1.2.1
 
