--- orig/Android.mk	2018-10-31 15:58:59.000000000 +0100
+++ patched/Android.mk	2019-03-08 12:11:35.271258500 +0100
@@ -3,7 +3,7 @@


 # Enable this if you want to support loading FLAC music with libFLAC
-SUPPORT_FLAC ?= true
+SUPPORT_FLAC := false
 FLAC_LIBRARY_PATH := external/flac-1.3.2

 # Enable this if you want to support loading OGG Vorbis music via Tremor
@@ -12,11 +12,11 @@
 VORBIS_LIBRARY_PATH := external/libvorbisidec-1.2.1

 # Enable this if you want to support loading MP3 music via MPG123
-SUPPORT_MP3_MPG123 ?= true
+SUPPORT_MP3_MPG123 := false
 MPG123_LIBRARY_PATH := external/mpg123-1.25.6

 # Enable this if you want to support loading MOD music via modplug
-SUPPORT_MOD_MODPLUG ?= true
+SUPPORT_MOD_MODPLUG := false
 MODPLUG_LIBRARY_PATH := external/libmodplug-0.8.9.0

 # Enable this if you want to support TiMidity
