diff --git a/pyogg/library_loader.py b/pyogg/library_loader.py
index c2ba36c..383331a 100644
--- a/pyogg/library_loader.py
+++ b/pyogg/library_loader.py
@@ -54,7 +54,7 @@ def load_other(name, paths = None):
                     except:
                         pass
     else:
-        for path in [os.getcwd(), _here]:
+        for path in [os.path.join(os.environ['ANDROID_PRIVATE'], '..', 'lib')]:
             for style in other_styles:
                 candidate = os.path.join(path, style.format(name))
                 if os.path.exists(candidate):
