diff -Naur twisted-twisted-19.7.0/src/twisted/python/_setup.py twisted-twisted-19.7.0_patched/src/twisted/python/_setup.py
--- twisted-twisted-19.7.0/src/twisted/python/_setup.py	2019-07-28 11:17:29.000000000 +0200
+++ twisted-twisted-19.7.0_patched/src/twisted/python/_setup.py	2019-10-21 22:10:03.643068863 +0200
@@ -282,7 +282,6 @@
     requirements = [
         "zope.interface >= 4.4.2",
         "constantly >= 15.1",
-        "incremental >= 16.10.1",
         "Automat >= 0.3.0",
         "hyperlink >= 17.1.1",
         "PyHamcrest >= 1.9.0",
@@ -291,8 +290,6 @@
 
     arguments.update(dict(
         packages=find_packages("src"),
-        use_incremental=True,
-        setup_requires=["incremental >= 16.10.1"],
         install_requires=requirements,
         entry_points={
             'console_scripts': _CONSOLE_SCRIPTS
