diff --git a/src/twisted/python/_setup.py b/src/twisted/python/_setup.py
index 32cb096c7..a607fef07 100644
--- a/src/twisted/python/_setup.py
+++ b/src/twisted/python/_setup.py
@@ -160,11 +160,6 @@ class ConditionalExtension(Extension, object):
 
 # The C extensions used for Twisted.
 _EXTENSIONS = [
-    ConditionalExtension(
-        "twisted.test.raiser",
-        sources=["src/twisted/test/raiser.c"],
-        condition=lambda _: _isCPython),
-
     ConditionalExtension(
         "twisted.internet.iocpreactor.iocpsupport",
         sources=[
