--- /home/tito/code/python-for-android/build/pil/Imaging-1.1.7/setup.py	2009-11-15 17:06:10.000000000 +0100
+++ Imaging-1.1.7/setup.py.tmpl	2011-04-20 16:59:04.000000000 +0200
@@ -5,7 +5,7 @@
 # Usage: python setup.py install
 #
 
-import glob, os, re, struct, string, sys
+import glob, os, re, string, sys
 
 # make it possible to run the setup script from another directory
 try:
@@ -34,10 +34,10 @@
 # TIFF_ROOT = libinclude("/opt/tiff")
 
 TCL_ROOT = None
-JPEG_ROOT = None
-ZLIB_ROOT = None
+JPEG_ROOT = "_LIBS_", "_JNI_/jpeg"
+ZLIB_ROOT = libinclude('_NDKPLATFORM_/usr')
 TIFF_ROOT = None
-FREETYPE_ROOT = None
+FREETYPE_ROOT = "_LIBS_", "_JNI_/freetype/include"
 LCMS_ROOT = None
 
 # FIXME: add mechanism to explicitly *disable* the use of a library
@@ -147,7 +147,6 @@
             add_directory(library_dirs, "/opt/local/lib")
             add_directory(include_dirs, "/opt/local/include")
 
-        add_directory(library_dirs, "/usr/local/lib")
         # FIXME: check /opt/stuff directories here?
 
         prefix = sysconfig.get_config_var("prefix")
@@ -207,12 +206,6 @@
             if os.path.isfile(os.path.join(tcl_dir, "tk.h")):
                 add_directory(include_dirs, tcl_dir)
 
-        # standard locations
-        add_directory(library_dirs, "/usr/local/lib")
-        add_directory(include_dirs, "/usr/local/include")
-
-        add_directory(library_dirs, "/usr/lib")
-        add_directory(include_dirs, "/usr/include")
 
         #
         # insert new dirs *before* default libs, to avoid conflicts
@@ -299,8 +292,6 @@
             defs.append(("HAVE_LIBZ", None))
         if sys.platform == "win32":
             libs.extend(["kernel32", "user32", "gdi32"])
-        if struct.unpack("h", "\0\1")[0] == 1:
-            defs.append(("WORDS_BIGENDIAN", None))
 
         exts = [(Extension(
             "_imaging", files, libraries=libs, define_macros=defs
