diff -ur Python-2.7.2.orig/Lib/ctypes/util.py Python-2.7.2/Lib/ctypes/util.py
--- Python-2.7.2.orig/Lib/ctypes/util.py	2011-06-11 16:46:24.000000000 +0100
+++ Python-2.7.2/Lib/ctypes/util.py	2015-05-10 15:50:18.906203529 +0100
@@ -71,7 +71,21 @@
     def find_library(name):
         return name
 
-if os.name == "posix" and sys.platform == "darwin":
+# this test is for android specifically shoudl match here and ignore any
+# of the other platform tests below
+if os.name == "posix":
+    def find_library(name):
+        """ hack to find librarys for kivy and android 
+        split the path and get the first parts which will give us 
+        the app path something like /data/data/org.app.foo/"""
+        app_root = os.path.abspath('../../').split(os.path.sep)
+        lib_search = os.path.sep.join(app_root) + os.path.sep + 'lib'
+        for filename in os.listdir(lib_search):
+            if filename.endswith('.so') and name in filename:
+                return lib_search + os.path.sep + filename
+        return None
+
+elif os.name == "posix" and sys.platform == "darwin":
     from ctypes.macholib.dyld import dyld_find as _dyld_find
     def find_library(name):
         possible = ['lib%s.dylib' % name,
Only in Python-2.7.2/Lib/ctypes: util.py.save
Only in Python-2.7.2/Lib/ctypes: util.py.save.1
