diff --git a/_setupares.py b/_setupares.py
index c42fe369..cd8854df 100644
--- a/_setupares.py
+++ b/_setupares.py
@@ -42,7 +42,7 @@ cflags = ('CFLAGS="%s"' % (cflags,)) if cflags else ''
 ares_configure_command = ' '.join([
     "(cd ", quoted_dep_abspath('c-ares'),
     " && if [ -r include/ares_build.h ]; then cp include/ares_build.h include/ares_build.h.orig; fi ",
-    " && sh ./configure --disable-dependency-tracking --disable-tests -C " + cflags,
+    " && sh ./configure --host={} --disable-dependency-tracking --disable-tests -C ".format(os.environ['COMMAND_PREFIX']) + cflags,
     " && cp src/lib/ares_config.h include/ares_build.h \"$OLDPWD\" ",
     " && cat include/ares_build.h ",
     " && if [ -r include/ares_build.h.orig ]; then mv include/ares_build.h.orig include/ares_build.h; fi)",
diff --git a/_setuplibev.py b/_setuplibev.py
index f05c2fe9..32f9bd81 100644
--- a/_setuplibev.py
+++ b/_setuplibev.py
@@ -28,7 +28,7 @@ LIBEV_EMBED = should_embed('libev')
 # Configure libev in place
 libev_configure_command = ' '.join([
     "(cd ", quoted_dep_abspath('libev'),
-    " && sh ./configure -C > configure-output.txt",
+    " && sh ./configure --host={} -C > configure-output.txt".format(os.environ['COMMAND_PREFIX']),
     ")",
 ])
 
