diff -Naur gevent-1.1.1/setup.py gevent-1.1.1_diff/setup.py
--- gevent-1.1.1/setup.py	2016-04-04 17:27:33.000000000 +0200
+++ gevent-1.1.1_diff/setup.py	2016-05-10 10:10:39.145881610 +0200
@@ -96,7 +96,7 @@
 # and the PyPy branch will clean it up.
 libev_configure_command = ' '.join([
     "(cd ", _quoted_abspath('libev/'),
-    " && /bin/sh ./configure ",
+    " && /bin/sh ./configure --host={}".format(os.environ['TOOLCHAIN_PREFIX']),
     " && cp config.h \"$OLDPWD\"",
     ")",
     '> configure-output.txt'
@@ -112,7 +112,7 @@
 # Use -r, not -e, for support of old solaris. See https://github.com/gevent/gevent/issues/777
 ares_configure_command = ' '.join(["(cd ", _quoted_abspath('c-ares/'),
                                    " && if [ -r ares_build.h ]; then cp ares_build.h ares_build.h.orig; fi ",
-                                   " && /bin/sh ./configure " + _m32 + "CONFIG_COMMANDS= CONFIG_FILES= ",
+                                   " && /bin/sh ./configure --host={} ".format(os.environ['TOOLCHAIN_PREFIX']) + "CONFIG_COMMANDS= CONFIG_FILES= ",
                                    " && cp ares_config.h ares_build.h \"$OLDPWD\" ",
                                    " && mv ares_build.h.orig ares_build.h)",
                                    "> configure-output.txt"])
