--- libtorrent/bindings/python/setup.py.orig	2018-11-26 22:21:48.772142135 +0100
+++ libtorrent/bindings/python/setup.py	2018-11-26 22:23:23.092141235 +0100
@@ -167,7 +167,7 @@
         extra_compile = flags.parse(extra_cmd)

         ext = [Extension(
-            'libtorrent',
+            'libtorrent_rasterbar',
             sources=sorted(source_list),
             language='c++',
             include_dirs=flags.include_dirs,
@@ -178,7 +178,7 @@
         ]

 setup(
-    name='python-libtorrent',
+    name='libtorrent',
     version='1.2.1',
     author='Arvid Norberg',
     author_email='arvid@libtorrent.org',
