--- boost/user-config.jam.orig	2018-12-07 14:16:45.911924859 +0100
+++ boost/user-config.jam	2018-12-07 14:20:16.243922853 +0100
@@ -9,6 +9,8 @@
 local PYTHON_INCLUDE = [ os.environ PYTHON_INCLUDE ] ;
 local PYTHON_LINK_VERSION = [ os.environ PYTHON_LINK_VERSION ] ;
 local PYTHON_MAJOR_MINOR = [ os.environ PYTHON_MAJOR_MINOR ] ;
+local OPENSSL_BUILD_PATH = [ os.environ OPENSSL_BUILD_PATH ] ;
+local OPENSSL_VERSION = [ os.environ OPENSSL_VERSION ] ;
 
 #using clang : $(ARCH) : $(ANDROID_BINARIES_PATH)/clang++ :
 #<archiver>$(ANDROID_BINARIES_PATH)/llvm-ar
@@ -56,6 +58,9 @@
 <linkflags>-Wl,-z,relro
 <linkflags>-Wl,-z,now
 <linkflags>-lc++_shared
+<linkflags>-L$(OPENSSL_BUILD_PATH)
+<linkflags>-lcrypto$(OPENSSL_VERSION)
+<linkflags>-lssl$(OPENSSL_VERSION)
 <linkflags>-L$(PYTHON_ROOT)
 <linkflags>-lpython$(PYTHON_LINK_VERSION)
 <linkflags>-Wl,-O1
