diff -r 9ecdf084933c setup.py
--- a/setup.py	Wed May 13 14:09:03 2015 +0100
+++ b/setup.py	Fri May 22 10:14:29 2015 +0100
@@ -14,8 +14,8 @@
 #no-download-t1-files=yes
 #ignore-system-libart=yes
 # if used on command line the config values are not used
-dlt1 = not specialOption('--no-download-t1-files')
-isla = specialOption('--ignore-system-libart')
+dlt1 = False
+isla = True
 
 try:
     import configparser
@@ -121,39 +121,6 @@
         else:
             P.insert(x, d)
 
-class inc_lib_dirs:
-    L = None
-    I = None
-    def __call__(self):
-        if self.L is None:
-            L = []
-            I = []
-            if platform == "cygwin":
-                aDir(L, os.path.join("/usr/lib", "python%s" % sys.version[:3], "config"))
-            elif platform == "darwin":
-                # attempt to make sure we pick freetype2 over other versions
-                aDir(I, "/sw/include/freetype2")
-                aDir(I, "/sw/lib/freetype2/include")
-                # fink installation directories
-                aDir(L, "/sw/lib")
-                aDir(I, "/sw/include")
-                # darwin ports installation directories
-                aDir(L, "/opt/local/lib")
-                aDir(I, "/opt/local/include")
-            aDir(I, "/usr/local/include")
-            aDir(L, "/usr/local/lib")
-            aDir(I, "/usr/include")
-            aDir(L, "/usr/lib")
-            aDir(I, "/usr/include/freetype2")
-            prefix = sysconfig.get_config_var("prefix")
-            if prefix:
-                aDir(L, pjoin(prefix, "lib"))
-                aDir(I, pjoin(prefix, "include"))
-            self.L=L
-            self.I=I
-        return self.I,self.L
-inc_lib_dirs=inc_lib_dirs()
-
 def getVersionFromCCode(fn):
     import re
     tag = re.search(r'^#define\s+VERSION\s+"([^"]*)"',open(fn,'r').read(),re.M)
@@ -244,11 +211,7 @@
         ]
 
 def get_fonts(PACKAGE_DIR, reportlab_files):
-    import sys, os, os.path, zipfile, io
-    if isPy3:
-        import urllib.request as ureq
-    else:
-        import urllib2 as ureq
+    import os, os.path
     rl_dir = PACKAGE_DIR['reportlab']
     if not [x for x in reportlab_files if not os.path.isfile(pjoin(rl_dir,x))]:
         infoline("Standard T1 font curves already downloaded")
@@ -257,6 +220,11 @@
         infoline('not downloading T1 font curve files')
         return
     try:
+        if isPy3:
+            import urllib.request as ureq
+        else:
+            import urllib2 as ureq
+        import zipfile, io
         infoline("Downloading standard T1 font curves")
 
         remotehandle = ureq.urlopen("http://www.reportlab.com/ftp/pfbfer-20070710.zip")
@@ -448,7 +416,8 @@
                 FT_LIB_DIR=[FT_LIB_DIR] if FT_LIB_DIR else []
                 FT_INC_DIR=config('FREETYPE_PATHS','inc')
                 FT_INC_DIR=[FT_INC_DIR] if FT_INC_DIR else []
-            I,L=inc_lib_dirs()
+            I=["_FT_INC_"]
+            L=["_FT_LIB_"]
             ftv = None
             for d in I:
                 if isfile(pjoin(d, "ft2build.h")):
