diff --git a/numpy/distutils/ccompiler.py b/numpy/distutils/ccompiler.py
index 6438790..a5f1527 100644
--- a/numpy/distutils/ccompiler.py
+++ b/numpy/distutils/ccompiler.py
@@ -686,13 +686,13 @@ def CCompiler_cxx_compiler(self):
         return self
 
     cxx = copy(self)
-    cxx.compiler_so = [cxx.compiler_cxx[0]] + cxx.compiler_so[1:]
+    cxx.compiler_so = cxx.compiler_cxx + cxx.compiler_so[1:]
     if sys.platform.startswith('aix') and 'ld_so_aix' in cxx.linker_so[0]:
         # AIX needs the ld_so_aix script included with Python
         cxx.linker_so = [cxx.linker_so[0], cxx.compiler_cxx[0]] \
                         + cxx.linker_so[2:]
     else:
-        cxx.linker_so = [cxx.compiler_cxx[0]] + cxx.linker_so[1:]
+        cxx.linker_so = cxx.compiler_cxx + cxx.linker_so[1:]
     return cxx
 
 replace_method(CCompiler, 'cxx_compiler', CCompiler_cxx_compiler)
