From: Mikhail Zakharov <mzakharo@gmail.com>
Date: Sun, 5 Jun 2022 11:14:23 -0400
Subject: [PATCH] fix missing threads.h

---
 numpy/f2py/cfuncs.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)


diff --git a/numpy/f2py/cfuncs.py b/numpy/f2py/cfuncs.py
index bdd27ad..39cb470 100644
--- a/numpy/f2py/cfuncs.py
+++ b/numpy/f2py/cfuncs.py
@@ -586,8 +586,7 @@ cppmacros["F2PY_THREAD_LOCAL_DECL"] = """\
    so `!defined(__STDC_NO_THREADS__)` may give false positive for the existence
    of `threads.h` when using an older release of glibc 2.12
    See gh-19437 for details on OpenBSD */
-#include <threads.h>
-#define F2PY_THREAD_LOCAL_DECL thread_local
+#define F2PY_THREAD_LOCAL_DECL __thread
 #elif defined(__GNUC__) \\
       && (__GNUC__ > 4 || (__GNUC__ == 4 && (__GNUC_MINOR__ >= 4)))
 #define F2PY_THREAD_LOCAL_DECL __thread
-- 
2.25.1

