commit 9a09edac303c534a38c5d829d8537176f8a8dfb9
Author: Alexander Taylor <alexanderjohntaylor@gmail.com>
Date:   Fri Jun 28 22:50:45 2019 +0100

    fix_environment_detection.patch

diff --git a/numpy/core/include/numpy/npy_common.h b/numpy/core/include/numpy/npy_common.h
index 64aaaac..e6293f9 100644
--- a/numpy/core/include/numpy/npy_common.h
+++ b/numpy/core/include/numpy/npy_common.h
@@ -164,12 +164,12 @@ extern long long __cdecl _ftelli64(FILE *);
     #endif
 #else
 #ifdef HAVE_FSEEKO
-    #define npy_fseek fseeko
+    #define npy_fseek fseek
 #else
     #define npy_fseek fseek
 #endif
 #ifdef HAVE_FTELLO
-    #define npy_ftell ftello
+    #define npy_ftell ftell
 #else
     #define npy_ftell ftell
 #endif
@@ -321,13 +321,15 @@ typedef unsigned char npy_bool;
 #define NPY_TRUE 1
 
 
-#if NPY_SIZEOF_LONGDOUBLE == NPY_SIZEOF_DOUBLE
-        typedef double npy_longdouble;
-        #define NPY_LONGDOUBLE_FMT "g"
-#else
-        typedef long double npy_longdouble;
-        #define NPY_LONGDOUBLE_FMT "Lg"
-#endif
+/* #if NPY_SIZEOF_LONGDOUBLE == NPY_SIZEOF_DOUBLE */
+/*         typedef double npy_longdouble; */
+/*         #define NPY_LONGDOUBLE_FMT "g" */
+/* #else */
+/*         typedef long double npy_longdouble; */
+/*         #define NPY_LONGDOUBLE_FMT "Lg" */
+/* #endif */
+typedef long double npy_longdouble;
+#define NPY_LONGDOUBLE_FMT "Lg"
 
 #ifndef Py_USING_UNICODE
 #error Must use Python with unicode enabled.
