diff --git a/numpy/distutils/system_info.py b/numpy/distutils/system_info.py
index bea120cf9..a448a83fc 100644
--- a/numpy/distutils/system_info.py
+++ b/numpy/distutils/system_info.py
@@ -719,6 +719,7 @@ class system_info(object):
         return self.get_paths(self.section, key)
 
     def get_libs(self, key, default):
+        return []
         try:
             libs = self.cp.get(self.section, key)
         except NoOptionError:
