diff -Naur  icu.py icu1.py
--- pyicu/icu.py	2012-11-23 21:28:55.000000000 +0100
+++ icu1.py	2016-05-14 14:45:44.160023949 +0200
@@ -34,4 +34,15 @@
 class InvalidArgsError(Exception):
     pass

+import ctypes
+import os
+root = os.environ["ANDROID_APP_PATH"]
+ctypes.cdll.LoadLibrary(os.path.join(root, "lib", "libgnustl_shared.so"))
+ctypes.cdll.LoadLibrary(os.path.join(root, "lib", "libicudata.so.57"))
+ctypes.cdll.LoadLibrary(os.path.join(root, "lib", "libicuuc.so.57"))
+ctypes.cdll.LoadLibrary(os.path.join(root, "lib", "libicui18n.so.57"))
+ctypes.cdll.LoadLibrary(os.path.join(root, "lib", "libicule.so.57"))
+del root
+del os
+
 from docs import *
