--- ./setup.py.orig	2017-07-31 15:20:01.380342826 +0300
+++ ./setup.py	2017-07-31 15:20:29.944475790 +0300
@@ -131,26 +131,35 @@

 elif "NDKPLATFORM" in environ:
     # enable python-for-android/py4a compilation
+
+    # ffmpeg:
     ffmpeg_lib, ffmpeg_include = get_paths('FFMPEG')
+    libraries.extend([
+        'avcodec', 'avdevice', 'avfilter', 'avformat',
+        'avutil', 'swscale', 'swresample', 'postproc', 'm'
+    ])
+    library_dirs.append(ffmpeg_lib)
+    include_dirs.append(ffmpeg_include)
+
+    # sdl:
     sdl_lib, sdl_include = get_paths('SDL')
+    if sdl_lib and sdl_include:
+        sdl = 'SDL2'
+        libraries.append(sdl)
+        library_dirs.append(sdl_lib)
+        include_dirs.append(sdl_include)
+    else:  # old toolchain
+        sdl = 'sdl'
+        libraries.append(sdl)
+        if sdl_lib: library_dirs.append(sdl_lib)
+        if sdl_include: include_dirs.append(sdl_include)

-    if not sdl_lib or not sdl_include:  # old toolchain
-        library_dirs = [ffmpeg_lib]
-        if sdl_lib:
-            library_dirs.append(sdl_lib)
-
-        include_dirs = [ffmpeg_include]
-        if sdl_include:
-            include_dirs.append(sdl_include)
-        sdl = "sdl"
-    else:
-        library_dirs = [ffmpeg_lib, sdl_lib]
-        include_dirs = [ffmpeg_include, sdl_include]
-        sdl = "SDL2"
-    libraries = ['avcodec', 'avdevice', 'avfilter', 'avformat',
-                 'avutil', 'swscale', 'swresample', 'postproc',
-                 'm',
-                 sdl]
+    # sdl2 mixer:
+    c_options['use_sdl2_mixer'] = c_options['use_sdl2_mixer'] and sdl == 'SDL2'
+    if c_options['use_sdl2_mixer']:
+        _, mixer_include = get_paths('SDL2_MIXER')
+        libraries.append('SDL2_mixer')
+        include_dirs.append(mixer_include)

 else:
