--- ./setup.py.orig	2016-11-28 23:59:57.000000000 +0300
+++ ./setup.py	2016-12-11 14:50:15.938008000 +0300
@@ -117,11 +117,13 @@
     include_dirs = [
         environ.get("SDL_INCLUDE_DIR"),
         environ.get("FFMPEG_INCLUDE_DIR")]
-    ffmpeg_libdir = environ.get("FFMPEG_LIB_DIR")
-    sdl = "SDL"
+    library_dirs = [
+        environ.get("SDL_LIB_DIR"),
+        environ.get("FFMPEG_LIB_DIR")]
+    sdl = "SDL2"
     libraries = ['avcodec', 'avdevice', 'avfilter', 'avformat',
                  'avutil', 'swscale', 'swresample', 'postproc',
-                 'sdl']
+                 'SDL2']
 
 else:
 
