diff -urN DoubleRatchet.ori/setup.py DoubleRatchet/setup.py
--- DoubleRatchet.ori/setup.py	2018-10-28 18:55:16.259540354 +0100
+++ DoubleRatchet/setup.py	2018-10-28 18:55:39.349541117 +0100
@@ -21,7 +21,7 @@
     author_email = "tim@cifg.io",
     license = "MIT",
     packages = find_packages(),
-    install_requires = [ "cryptography>=1.7.1" ],
+    install_requires = [],
     python_requires  = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, <4",
     zip_safe = True,
     classifiers = [
