diff --git a/jnius/jnius_jvm_android.pxi b/jnius/jnius_jvm_android.pxi
index ac89fec..71daa43 100644
--- a/jnius/jnius_jvm_android.pxi
+++ b/jnius/jnius_jvm_android.pxi
@@ -1,5 +1,5 @@
 # on android, rely on SDL to get the JNI env
-cdef extern JNIEnv *SDL_ANDROID_GetJNIEnv()
+cdef extern JNIEnv *WebView_AndroidGetJNIEnv()
 
 cdef JNIEnv *get_platform_jnienv():
-    return SDL_ANDROID_GetJNIEnv()
+    return <JNIEnv*>WebView_AndroidGetJNIEnv()
diff --git a/setup.py b/setup.py
index 740510f..0c8e55f 100644
--- a/setup.py
+++ b/setup.py
@@ -53,7 +53,7 @@ except ImportError:
 
 if PLATFORM == 'android':
     # for android, we use SDL...
-    LIBRARIES = ['sdl', 'log']
+    LIBRARIES = ['main', 'log']
     LIBRARY_DIRS = ['libs/' + getenv('ARCH')]
 elif PLATFORM == 'darwin':
     import subprocess
