diff --git a/jnius/jnius_jvm_android.pxi b/jnius/jnius_jvm_android.pxi
index ac89fec..71daa43 100644
--- a/jnius/jnius_jvm_android.pxi
+++ b/jnius/jnius_jvm_android.pxi
@@ -1,5 +1,5 @@
 # on android, rely on SDL to get the JNI env
-cdef extern JNIEnv *SDL_ANDROID_GetJNIEnv()
+cdef extern JNIEnv *WebView_AndroidGetJNIEnv()
 
 cdef JNIEnv *get_platform_jnienv():
-    return SDL_ANDROID_GetJNIEnv()
+    return <JNIEnv*>WebView_AndroidGetJNIEnv()
diff --git a/jnius/env.py b/jnius/env.py
--- a/jnius/env.py
+++ b/jnius/env.py
@@ -185,10 +185,10 @@ except ImportError:
 
 def get_libraries(platform):
     if platform == 'android':
         # for android, we use SDL...
-        return ['sdl', 'log']
+        return ['main', 'log']
 
     elif platform == 'win32':
         return ['jvm']

