diff --git a/setup.py b/setup.py
index b3c422d..f90694c 100644
--- a/setup.py
+++ b/setup.py
@@ -13,10 +13,11 @@ PY3 = sys.version_info >= (3,0,0)
 def getenv(key):
     val = environ.get(key)
     if val is not None:
-        if PY3:
-            return val.decode()
-        else:
-            return val
+        return val
+        # if PY3:
+        #     return val.decode()
+        # else:
+        #     return val
     else:
         return val
 
