--- zeroconf.orig/setup.py	2015-07-11 21:55:09.000000000 +0200
+++ zeroconf/setup.py	2017-02-23 01:04:13.370018716 +0100
@@ -55,12 +55,5 @@
         'mDNS',
     ],
     install_requires=[
-        'enum-compat',
-        # netifaces 0.10.5 has a bug that results in all interfaces' netmasks
-        # to be 255.255.255.255 on Windows which breaks things. See:
-        # * https://github.com/jstasiak/python-zeroconf/issues/84
-        # * https://bitbucket.org/al45tair/netifaces/issues/39/netmask-is-always-255255255255
-        'netifaces<=0.10.4',
-        'six',
     ],
 )
