--- Python-3.11.5/Lib/ctypes/util.py	2023-08-24 17:39:18.000000000 +0530
+++ Python-3.11.5.mod/Lib/ctypes/util.py	2023-11-18 22:12:17.356160615 +0530
@@ -4,7 +4,15 @@
 import sys
 
 # find_library(name) returns the pathname of a library, or None.
-if os.name == "nt":
+
+# This patch overrides the find_library to look in the right places on
+# Android
+if True:
+    from android._ctypes_library_finder import find_library as _find_lib
+    def find_library(name):
+        return _find_lib(name)
+
+elif os.name == "nt":
 
     def _get_build_version():
         """Return the version of MSVC that was used to build Python.
