--- Python-3.7.1/setup.py.orig	2018-10-20 08:04:19.000000000 +0200
+++ Python-3.7.1/setup.py	2019-02-17 00:24:30.715904412 +0100
@@ -1410,7 +1410,8 @@ class PyBuildExt(build_ext):
         if zlib_inc is not None:
             zlib_h = zlib_inc[0] + '/zlib.h'
             version = '"0.0.0"'
-            version_req = '"1.1.3"'
+            version_req = '"{}"'.format(
+                os.environ.get('ZLIB_VERSION', '1.1.3'))
             if host_platform == 'darwin' and is_macosx_sdk_path(zlib_h):
                 zlib_h = os.path.join(macosx_sdk_root(), zlib_h[1:])
             with open(zlib_h) as fp:
