diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -67,4 +67,11 @@
                 return fname
         return None
 
+# This patch overrides the find_library to look in the right places on
+# Android
+if True:
+    from android._ctypes_library_finder import find_library as _find_lib
+    def find_library(name):
+        return _find_lib(name)
+
 elif os.name == "posix" and sys.platform == "darwin":
